### ISSUE TITLE
[Feature] A flag to enforce consistency of unplugged files and node_modules

### ISSUE THREAD

**Issue by mekwall** [yarnpkg/berry#3210]
- [ ] I'd be willing to implement this feature ([contributing guide](https://yarnpkg.com/advanced/contributing))
- [x] This feature is important to have in this repository; a contrib plugin wouldn't do

**Describe the user story**

As a developer I sometimes manually delete dependencies and there's currently no easy way to tell yarn to reinstall that dependency. This is especially a problem when using nm linker and usually not with unplugged packages.

**Describe the solution you'd like**

I suggest adding a flag to `install` that disregard/recreates the entries in `.yarn-state.yml` so that the deleted package is installed again.

**Describe the drawbacks of your solution**

I don't think there are any drawbacks to adding this flag.

**Comment by larixer** [comment]
One thought in this direction is that pnp linker and nm linker behaviour are slightly different now in this regard. When you delete a folder of unplugged package, it gets reinstalled on the next install, but when you delete a folder inside node_modules it is not. I can experiment and make the behaviour of nm linker match the pnp linker - i.e. recreate deleted folders inside node_modules on the next install, but only if it will not have serious effects on nm linker performance.

**Comment by arcanis** [comment]
Yeah. I'm not convinced by the flag idea, since it adds a CLI complexity that will be rarely needed (almost never in the case of the PnP linker?), and essentially would do the same thing as a `rm` of the n_m folder would do. If you edit it yourself, then requiring you to erase your own changes afterwards seems fair.

**Comment by flacerdk** [comment]
@arcanis having to keep track of my changes so I can change them back is a lot less convenient than just reinstalling the package.

I do this a lot when debugging dependencies, and currently it doesn't seem like yarn 2 offers me a way to reinstall the dependency other than removing all of `node_modules` and starting over, which is also not very convenient. This is something I don't need to do with yarn 1.

**Comment by larixer** [comment]
@flacerdk We have different experience with yarn 1, it seems. With yarn 1 I had to `rm -rf node_modules && yarn` each time I wanted to revert my changes inside `node_modules`.
